core/client/src/main/java/alluxio/client/block/UnderStoreBlockInStream.java

```diff
@@ -59,10 +65,11 @@
    */
   public interface UnderStoreStreamFactory extends AutoCloseable {
     /**
+     * @param length the length to read from the file (set to Long.MAX_VALUE if unknown)
      * @return an input stream to under storage
      * @throws IOException if an IO exception occurs
      */
-    InputStream create() throws IOException;
+    InputStream create(long length) throws IOException;
```

Comment:
yes, it is easy to keep both. But I think it is better to have fewer interfaces to keep the code clean if there is no good reason.